lms/djangoapps/grades/services.py

```diff
@@ -69,23 +67,17 @@ def override_subsection_grade(self, user_id, course_key_or_id, usage_key_or_id,
         course_key = _get_key(course_key_or_id, CourseKey)
         usage_key = _get_key(usage_key_or_id, UsageKey)
 
-        grade = PersistentSubsectionGrade.objects.get(
+        grade = PersistentSubsectionGrade.read_grade(
             user_id=user_id,
-            course_id=course_key,
             usage_key=usage_key
         )
 
-        # Create override that will prevent any future updates to grade
-        override, _ = PersistentSubsectionGradeOverride.objects.update_or_create(
-            grade=grade,
-            earned_all_override=earned_all,
-            earned_graded_override=earned_graded
-        )
-
-        _ = PersistentSubsectionGradeOverrideHistory.objects.create(
-            override_id=override.id,
+        override = PersistentSubsectionGradeOverride.update_or_create_override(
+            requesting_user=None,
+            subsection_grade_model=grade,
             feature=PersistentSubsectionGradeOverrideHistory.PROCTORING,
```

Comment:
Is this always `PROCTORING`?

Reply:
Also, why does the core grades model layer need to know about all high-level features that use it?  Should the enum of features belong elsewhere?  Or can they just be namespaced string values?

Reply:
Right now, proctoring is the only thing that uses this service to do grade overrides.  I think the right thing to do (subsequent to this PR) is to actually remove the restricted set of choices for the `feature` field of the Override History model.  Any client that needs to use that code can be in charge of determining an appropriate value for that field.  So 'proctoring' would be an enum specified by the proctoring library, and 'gradebook' would be an enum specified...somewhere - gradebook isn't an installed library, so maybe that enum needs to live somewhere in the grades app, or maybe the instructor dashboard app?